feature/container/containerz/tests/container_lifecycle/containerz_test.go

```diff
@@ -2,44 +2,53 @@ package container_lifecycle_test
 
 import (
 	"context"
-	"crypto/tls"
 	"flag"
 	"testing"
 	"time"
 
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/ondatra"
+
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/go-cmp/cmp/cmpopts"
 	"github.com/openconfig/containerz/client"
-	cpb "github.com/openconfig/featureprofiles/internal/cntrsrv/proto/cntr"
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/credentials"
-	"google.golang.org/grpc/credentials/insecure"
 )
 
 var (
 	containerTar        = flag.String("container_tar", "/tmp/cntrsrv.tar", "The container tarball to deploy.")
 	containerUpgradeTar = flag.String("container_upgrade_tar", "/tmp/cntrsrv-upgrade.tar", "The container tarball to upgrade to.")
-	containerzAddr      = flag.String("containerz_addr", "localhost:19999", "containerz server address")
 )
 
 const (
 	instanceName = "test-instance"
 )
 
-func containerzClient(ctx context.Context, t *testing.T) *client.Client {
-	client.Dial = func(ctx context.Context, target string, opts ...grpc.DialOption) (conn *grpc.ClientConn, err error) {
-		return grpc.DialContext(ctx, target, grpc.WithTransportCredentials(insecure.NewCredentials()))
-	}
-	cli, err := client.NewClient(ctx, *containerzAddr)
-	if err != nil {
-		t.Fatalf("unable to dial containerz: %v", err)
-	}
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
 
-	return cli
+func containerzClient(ctx context.Context, t *testing.T) *client.Client {
+	dut := ondatra.DUT(t, "dut")
+
+	dut.Config().New().WithAristaText(`
+	management api gnoi
+  	service containerz
+      transport gnmi default
+      !
+      container runtime
+         vrf default
+			!
+	`).Append(t)
+
+	t.Logf("Waiting for device to ingest its config.")
+	time.Sleep(time.Minute)
+
+	return client.NewClientFromStub(dut.RawAPIs().GNOI(t).Containerz())
 }
 
 func startContainer(ctx context.Context, t *testing.T) *client.Client {
 	cli := containerzClient(ctx, t)
+	cli.RemoveContainer(ctx, instanceName, true)
```

Comment:
Suggest to add a comment as to why this is needed (otherwise it's a bit odd that we remove a container when we are calling a fn to start a container).